### PR TITLE
Hubspot tiering traits

### PR DESCRIPTION
**Pull request by jordanh**
Implements #1790 , which will allow us to easily compute metrics and manage marketing automations for paying vs. non-paying users

Some notes:

- I added some crummy unit tests
- I added a migration to index on the org tier
- The way we call Identify has been refactored into a helper to keep trait passing consistent
- We now call identify in more places, to attempt to keep the traits updated when they change (it doesn't have to be _perfect,_ but I think what I've got now is a reasonable start)

### CODE REVIEW

**Comment by jordanh** [comment]
# Test Template
## Status: in-progress/complete

## Tests

- [ ] new migration runs `$ npm run db:migrate`

### Making sure the `SEGMENT_WRITE_KEY` environment variable is set to the developer key I can...

- [ ] Log in
- [ ] Go to my user's profile page
- [ ] Change my name
- [ ] Login to https://app.segment.com/parabol-jordan/sources/actiondevelopment/debugger and observe the identify call segment event
- [ ] See the new traits `tierPersonalCount`, `tierProBillingLeaderCount`, `tierProCount`, and `parabolId`

File: src/server/graphql/queries/helpers/countTiersForUserId.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      {
+        tierPersonalCount: 0,
+        tierProCount: 0,
+        tierProBillingLeaderCount: 0
+      },
+      (acc, org) => ({
+        tierPersonalCount: r.add(
+          acc('tierPersonalCount'),
+          r.branch(
+            org('tier').default(PERSONAL).eq(PERSONAL),
```
Comment (jordanh) [inline src/server/graphql/queries/helpers/countTiersForUserId.js]: Our mock team data doesn't have the tier set, hence the default

File: src/server/graphql/rootQuery.js [hunk 2]
```diff
@@ -3,12 +3,14 @@ import User from 'server/graphql/types/User';
...
 import activeProOrgCount from 'server/graphql/queries/activeProOrgCount';
 import activeProUserCount from 'server/graphql/queries/activeProUserCount';
+import countTiersForUser from 'server/graphql/queries/countTiersForUser';
 
 export default new GraphQLObjectType({
   name: 'Query',
   fields: () => ({
     activeProOrgCount,
     activeProUserCount,
+    countTiersForUser,
```
Comment (jordanh) [inline src/server/graphql/rootQuery.js]: I made it so we can run this as an ad-hoc query, too

File: src/server/graphql/queries/helpers/countTiersForUserId.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import {PERSONAL, PRO} from 'universal/utils/constants';
+
+// breaking this out into its own helper so it can be used directly to
+// populate segment traits
+
+const countTiersForUserId = (userId) => {
+  const r = getRethink();
+
+  return r.table('Organization')
+    .getAll(userId, {index: 'orgUsers'})
```
Comment (mattkrick) [inline src/server/graphql/queries/helpers/countTiersForUserId.js]: Inactive is a trait assigned to a user, and is then denormalized onto the org users. Might be cleaner to just filter out here.

File: src/server/utils/sendSegmentEvent.js [hunk 4]
```diff
@@ -11,7 +12,8 @@ const getTraits = (userIds) => {
       createdAt: r.row('createdAt').default(0),
       email: r.row('email').default(''),
       id: r.row('id').default(''),
-      name: r.row('preferredName').default('')
+      name: r.row('preferredName').default(''),
+      parabolId: r.row('id').default('') // passed as a distinct trait name for HubSpot
```
Comment (mattkrick) [inline src/server/utils/sendSegmentEvent.js]: Silly question, why can't you use the already existing id field?

File: src/server/billing/helpers/adjustUserCount.js [hunk 5]
```diff
@@ -121,4 +123,6 @@ export default async function adjustUserCount(userId, orgInput, type) {
   }, []);
 
   await Promise.all(stripePromises);
+  // publish any changes to user traits (like tier counts) to segment:
+  await sendSegmentIdentify(userId);
```
Comment (mattkrick) [inline src/server/billing/helpers/adjustUserCount.js]: No need to await segment things, but it's cool if you wanna keep it

File: src/server/utils/sendSegmentEvent.js (hunk 4 above)
Comment (jordanh) [reply src/server/utils/sendSegmentEvent.js]: It's actually interesting, or lame, depending on your love for Segment and HubSpot.

Segment will publish its traits to HubSpot as HubSpot Contact Properties. However, there are some reserved words and `id` is one of them. So `parabolId` it is! (Which will actually be xformed to snake-case as `parabol_id` by the Segment HubSpot integration).

It's all duct-tape and bailing wire.

File: src/server/billing/helpers/adjustUserCount.js (hunk 5 above)
Comment (jordanh) [reply src/server/billing/helpers/adjustUserCount.js]: Oh true. I'll leave it for now, but if we notice anything laggy we'll know where to start :)

**Pull request merged**